### DISCUSSION TITLE
bug: Cannot filter events by argument with useContractEvent

### DISCUSSION THREAD

**Discussion by cosiato** [wevm/wagmi#2455] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.0.5

### Current Behavior

We cannot directly use a React state variable inside the callback function of the `useContractEvent` hook to filter event logs.

It would be great to be able to scope by event arguments using `useContractEvent` like described [here](https://viem.sh/docs/contract/watchContractEvent.html#scoping-to-event-arguments)

### Expected Behavior

Being able to filter event logs dynamically

**Comment by nfts2me** [comment]
We'd also love this to be added!

**Comment by SahilPanhotra** [comment]
Any updates on this?
